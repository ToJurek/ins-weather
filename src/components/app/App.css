.App {
  text-align: center;
  height: 100vh;
  background-position: bottom;
  display: flex;
  justify-content: center;
  align-items: center;
}

