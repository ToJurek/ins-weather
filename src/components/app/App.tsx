import React from 'react';
import './App.css';
import {Afternoon, Morning, Night} from '../../images/background/export';
import Weather from "../weather/WeatherComponent";


function getBackground(): string {
    let hour: number = new Date().getHours();
    if (hour >= 6 && hour < 11)
        return Morning;
    if (hour >= 11 && hour < 19)
        return Afternoon;
    return Night
}

function App() {
    let style = {backgroundImage: `url(${getBackground()})`};
    return (
        <div className="App" style={style}>
            <Weather/>
        </div>
    );
}

export default App;
