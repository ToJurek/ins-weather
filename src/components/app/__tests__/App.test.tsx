import React from "react";
import Enzyme, {mount} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import App from "../App";
Enzyme.configure({adapter: new Adapter()});

describe('App', function () {
    test('should match to snapshot', () => {
        let wrapper = mount(<App/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});
