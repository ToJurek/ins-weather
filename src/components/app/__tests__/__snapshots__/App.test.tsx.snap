// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match to snapshot 1`] = `
<App>
  <div
    className="App"
    style={
      Object {
        "backgroundImage": "url(morning.jpg)",
      }
    }
  >
    <Weather>
      <div
        className="shadow-background"
      >
        <div
          className="card"
        >
          <State
            error=""
            isLoaded={false}
          >
            <div
              className="spinner"
            >
              <div
                className="bounce1"
              />
              <div
                className="bounce2"
              />
              <div
                className="bounce3"
              />
            </div>
          </State>
        </div>
      </div>
    </Weather>
  </div>
</App>
`;
